src/progress.rs

```diff
@@ -196,12 +196,15 @@ pub fn snapshot_progress(ui: &mut Ui) -> Option<impl Fn(&RepoPath) + '_> {
                     .output_guard(format!("\r{}", Clear(ClearType::CurrentLine))),
             );
         }
-        _ = write!(
-            state.ui,
-            "\r{}Snapshotting {}",
-            Clear(ClearType::CurrentLine),
-            path.to_fs_path(Path::new("")).display()
-        );
+
+        let bar_width = state.ui.term_width().map(usize::from).unwrap_or(20);
```

Comment:
20 felt like a reasonable value that would capture the "Snapshotting" part and a few characters from the path, but I'm open to suggestions...

Reply:
Six characters of the path seems of pretty limited. use. How about `80`, as a typical terminal width? Note also this path only runs when stdout is a tty so I doubt this will fail often.

Reply:
Done.